src/test/rgw/test_store_bench.cc

```diff
@@ -0,0 +1,281 @@
+	rgw::sal::Attrs attrs;
+	bufferlist etag_bl;
+
+	// time prepare->process->complete
+	std::chrono::steady_clock::time_point t1 = std::chrono::steady_clock::now();
+	etag_bl.append(etag.c_str(), etag.size());
+	attrs.emplace(RGW_ATTR_ETAG, etag_bl);
+	obj->set_obj_attrs(CONF.dpp, &attrs, nullptr, null_yield);
+	processor = CONF.store->get_atomic_writer(
+			CONF.dpp, null_yield, std::move(obj), Config::instance().uid, &pr, 1, "");
+	processor->prepare(null_yield);
+	auto proc_ret = processor->process(std::move(*bl), 0);
+	if (proc_ret < 0) {
+		std::cerr << "Error processor process " << proc_ret << std::endl;
+	}
+	auto time = real_time();
+	char *if_match = NULL;
+	char *if_nomatch = NULL;
+	std::string user_data;
+	rgw_zone_set zs;
+	bool pcanceled = false;
+	int op_ret = processor->complete(obj_size, "", &time, real_time(), attrs,
+																	 real_time(), if_match, if_nomatch,
+																	 &user_data, &zs, &pcanceled, null_yield);
+	if (op_ret < 0) {
+		std::cerr << "Error processor complete " << op_ret << std::endl;
+	}
+	std::chrono::steady_clock::time_point t2 = std::chrono::steady_clock::now();
+	return std::chrono::duration_cast<std::chrono::milliseconds>((t2 - t1))
+			.count();
+}
+
+int main(int argc, char **argv) {
+	std::map<std::string, std::string> defaults = {
+			{"debug_rgw", "1/5"},
+			{"keyring", "$rgw_data/keyring"},
+			{"objecter_inflight_ops", "24576"},
+			// require a secure mon connection by default
+			{"ms_mon_client_mode", "secure"},
+			{"auth_client_required", "cephx"}};
+	int flags = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+	flags |= CINIT_FLAG_DEFER_DROP_PRIVILEGES;
+	auto args = argv_to_vec(argc, argv);
+	auto cct = rgw_global_init(&defaults, args, CEPH_ENTITY_TYPE_CLIENT,
+														 CODE_ENVIRONMENT_DAEMON, flags);
+
+	auto args_copy = args;
+	std::string val;
+	int pow_2_increments = 1;
+	size_t iterations = 1;
+	for (std::vector<const char *>::iterator i = args_copy.begin();
+			 i != args_copy.end();) {
+		if (ceph_argparse_witharg(args_copy, i, &val, "--pow-2-increments",
+															(char *)NULL)) {
+			pow_2_increments = std::stoul(val);
+		} else if (ceph_argparse_witharg(args_copy, i, &val, "--iterations",
+																		 (char *)NULL)) {
+			iterations = std::stoul(val);
+		} else {
+			++i;
+		}
+	}
+	if (pow_2_increments < 0) {
+		std::cerr << "Wrong number of increments: " << pow_2_increments << std::endl;
+		exit(1);
+	}
+	std::cout << "number of power of 2 increments = " << pow_2_increments << std::endl;
+	std::cout << "iterations = " << iterations << std::endl;
+	DoutPrefix dp(cct.get(), 1, "test");
+	DoutPrefixProvider *dpp = dynamic_cast<DoutPrefixProvider *>(&dp);
+	rgw::sal::Store *store = StoreManager::get_storage(
+			dpp, cct.get(), "rados", false, false, false, false, false);
+
+	if (store == nullptr) {
+		std::cerr << "get raw storage failed" << std::endl;
+		exit(1);
+	}
+	RGWRados *rados = ((rgw::sal::RadosStore *)store)->getRados();
```

Comment:
With the two changes above, there should be no need to cast store to RadosStore, and this should work with other stores like DBStore (which is a goal)

Reply:
True.